File: js/src/program/ops/opConstant.js
export default {
  /** { value: any } */
  op(runtime, input, params, scope, next) {
    return next(params.value ?? null, scope);
  },

  /** { value: any } */
  map(runtime, params) {
    return {
      value: runtime.normalizeValue(params.value),
    };
  },
};

File: go/program/ops/opConstant.go
package ops

import (
	"github.com/2manyvcos/jpl/go/definition"
	"github.com/2manyvcos/jpl/go/jpl"
	"github.com/2manyvcos/jpl/go/library"
)

type opConstant struct{}

// { value: any }
func (opConstant) OP(runtime jpl.JPLRuntime, input any, params definition.JPLInstructionParams, scope jpl.JPLRuntimeScope, next jpl.JPLScopedPiper) ([]any, jpl.JPLError) {
	return next.Pipe(params.Value, scope)
}

// { value: any }
func (opConstant) Map(runtime jpl.JPLRuntime, params jpl.JPLInstructionParams) (result definition.JPLInstructionParams, err jpl.JPLError) {
	if result.Value, err = library.NormalizeValue(params.Value); err != nil {
		return
	}
	return
}
